poucave/html/js/app.js

```diff
@@ -0,0 +1,627 @@
+            const tooltip = document.createElementNS("http://www.w3.org/2000/svg", "title");
+            tooltip.textContent = `${c.project}/${c.name}:\n${c.description}`;
+            indicator.appendChild(tooltip);
+
+            indicator.addEventListener("click", () => {
+              document.getElementById(`check--${c.project}--${c.name}`).scrollIntoView({
+                behavior: "smooth",
+                block: "center",
+              });
+            });
+          }
+
+          indicator.removeAttribute("fill");
+          indicator.classList.remove("fill-gray", "fill-red", "fill-green");
+          let fillClass = "fill-gray"
+          if (!r.isLoading) {
+            fillClass = r.success ? "fill-green" : "fill-red";
+          }
+          indicator.classList.add(fillClass);
+        }
+      }
+    });
+  }
+
+  render() {
+    const { diagramHidden, diagramReady } = this.state;
+    const { results } = this.props;
+    const diagramClass = diagramReady ? "" : "invisible";
+    const diagramCardClass = diagramHidden ? "d-none" : "";
+
+    let isLoading = false;
+    Object.keys(results).forEach(k => {
+      const r = results[k];
+      isLoading |= r.isLoading;
+    });
+
+    let loader = null;
+    if (isLoading) {
+      loader = html`<i class="fa fa-spinner fa-spin" />`;
+    }
+
+    return html`
+      <div class="card ${diagramCardClass}">
+        <div class="card-header">
+          <h4 class="card-title">System Overview</h4>
+          <div class="card-options">${loader}</div>
+        </div>
+        <div class="card-body">
+          <object 
+            ref="${this.svgRef}"
+            onload="${this.handleObjectLoad}"
+            onerror="${this.handleObjectError}"
+            class="d-block mx-auto system-diagram ${diagramClass}" 
+            data="/diagram.svg" 
+            type="image/svg+xml" 
+          />
+        </div>
+      </div>
+    `;
+  }
+}
+
+class Project extends Component {
+  renderStatus() {
+    const { checks } = this.props;
+
+    let isLoading = false;
+    let isHealthy = true;
+
+    checks.forEach(c => {
+      isLoading |= c.result.isLoading;
+      isHealthy &= c.result.success;
+    });
+
+    let color = "bg-gray";
+    let status = "loading";
+    if (!isLoading) {
+      color = isHealthy ? "bg-green" : "bg-red";
+      status = isHealthy ? "Healthy" : "Unhealthy";
+    }
+
+    return html`
+      <span class="badge ${color}">${status}</span>
+    `;
+  }
+
+  render({ name, checks, fetchCheckResult }) {
+    return html`
+      <section class="project mt-3 pt-4">
+        <div class="float-right mt-1 lh-1">
+          ${this.renderStatus()}
+        </div>
+        <h3 class="mb-4">
+          <i class="fa fa-layer-group mr-2"></i>
+          <span class="project-name">${name}</span>
+        </h3>
+        <div class="project-cards mb-1">
+          ${checks.map(c => html`<${Check} data="${c.data}" result="${c.result}" fetchResult="${fetchCheckResult}" />`)}
```

Comment:
Maybe keep the name `fetchCheckResult` so that we can follow the reference more easily